url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/StdArgList.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Scripty Programming Language
+ * Copyright (C) 2010-2012 Bruno Ranschaert, S.D.I.-Consulting BVBA
+ * http://www.sdi-consulting.be
+ * mailto://info@sdi-consulting.be
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.spec.args;
+
+import com.sdicons.scripty.parser.IContext;
+import com.sdicons.scripty.parser.Pair;
+
+/**
+ * There are 3 types of parameters in a standard argument list.
+ * 1. Fixed and required arguments. Each with its own type.
+ * 2. Optional. Each argument has its own type. These cannot be of type Pair, because this might conflict with the named arguments.
+ *    The optional parameters can have a default value which will be used when the argument is not present.
+ * 3. Named (optional or required). These are pairs at the end of the command line.
+ *    The named parameters can have a default value.
+ */
+public class StdArgList 
+implements IArgList
+{
+    public static final IArgList NOARG = new StdArgList(new FixedArg[]{}, new OptionalArg[]{}, new NamedArg[]{});
+    
+    private FixedArg req[];
+    private OptionalArg[] opt;
+    private NamedArg[] named;
+    
+    public StdArgList(FixedArg aReq[], OptionalArg[] aOpt, NamedArg[] aNamed)
+    {
+        req = aReq;
+        opt = aOpt;
+        named = aNamed;
+    }
+    
+    /**
+     * The ars are expected to have the form ("CMD", arg-1, arg-2, ... arg-n).
+     */
+    public Object[] guard(Object[] aArgs, IContext aCtx) 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 90-118 into a separate method.